### PR TITLE
feat(step): update KDE Plasmoids using libplasmoid-updater

### PR DESCRIPTION
**Pull request by uwuclxdy**
## What does this PR do

Adds integration with `libplasmoid-updater` to update KDE Plasma components (plasmoids) user and system-wide.
Fixes https://github.com/topgrade-rs/topgrade/issues/1020

It can also be used as a standalone program for those that wish to test (or just use) the library itself: https://github.com/uwuclxdy/plasmoid-updater

## Standards checklist

- [x] The PR title is descriptive
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself
- [x] If this PR introduces new user-facing messages they are translated

## For new steps

- [x] *Optional:* Topgrade skips this step where needed (**needs testing**)
- [x] *Optional:* The `--dry-run` option works with this step
- [x] *Optional:* The `--yes` option works with this step if it is supported by
  the underlying command

My best try at this ig. Needs testing (edge cases, error handling). As far as library is concerned, the code is a messy and worse I'd like it to be, but it works! It successfully finds and updates components that either Discover or Apdatify sometimes doesn't find.

### CODE REVIEW

File: src/steps/os/linux.rs [hunk 1]
```diff
@@ -1172,6 +1172,68 @@ pub fn run_cinnamon_spices_updater(ctx: &ExecutionContext) -> Result<()> {
...
+            println!("{e:#?}");
+            return Err(StepFailed.into());
+        }
+        Ok(result) => {
+            if result.available_updates.is_empty() {
+                // libplasmoid-updater already prints "no updates available" by this point
+                return Ok(());
+            }
+        }
+    };
```
Comment (uwuclxdy) [inline src/steps/os/linux.rs]: could probably remove this because `update()` checks for updates and prints the same "no updates available."

we're basically doing the same thing twice if updates are found.

**Comment by GideonBear** [comment]
FYI the MSRV check in CI will fail because of the current MSRV of 1.84 which will be bumped to 1.88 in v17 (which I will release soon, I hope). You don't have to change the edition and you can ignore the MSRV check CI for now.
`plasmoid-updater` does compile with Rust 1.88, I checked

**Comment by uwuclxdy** [comment]
@GideonBear yeah, i fixed the ordering problem tho. they were ordered but not in the right place in proportion to the entire enum

**Comment by uwuclxdy** [comment]
**before:**
https://asciinema.org/a/5yUWkvgGfgczGTs5

**after:**
https://asciinema.org/a/vJeRALgInwaWbQR2

File: Cargo.toml [hunk 2]
```diff
@@ -80,6 +80,9 @@ assets = [
   ["deployment/deb/_topgrade", "usr/share/zsh/vendor-completions/", "644"],
 ]
 
+[target.'cfg(target_os = "linux")'.dependencies]
+libplasmoid-updater = { version = "0.1.3", features = ["cli"] }
```
Comment (GideonBear) [inline Cargo.toml]: Why the CLI feature?

File: src/steps/os/linux.rs [hunk 3]
```diff
@@ -1172,6 +1172,46 @@ pub fn run_cinnamon_spices_updater(ctx: &ExecutionContext) -> Result<()> {
     ctx.execute(cinnamon_spice_updater).arg("--update-all").status_checked()
 }
 
+#[cfg(target_os = "linux")]
```
Comment (GideonBear) [inline src/steps/os/linux.rs]: Unnecessary since the whole module is under `#[cfg(target_os = "linux")]` in `mod.rs`

File: src/steps/os/linux.rs [hunk 4]
```diff
@@ -1172,6 +1172,46 @@ pub fn run_cinnamon_spices_updater(ctx: &ExecutionContext) -> Result<()> {
...
+pub fn run_plasmoid_updater(ctx: &ExecutionContext, system: bool, step_name: &str) -> Result<()> {
+    use crate::error::StepFailed;
+    use libplasmoid_updater::{update, Config as PlasmoidConfig};
```
Comment (GideonBear) [inline src/steps/os/linux.rs]: Can these just be global imports?

File: src/steps/os/linux.rs [hunk 5]
```diff
@@ -1172,6 +1172,46 @@ pub fn run_cinnamon_spices_updater(ctx: &ExecutionContext) -> Result<()> {
...
+
+    let with_yes = ctx.config().yes(Step::Plasmoids) || ctx.config().yes(Step::PlasmoidsSystem);
```
Comment (GideonBear) [inline src/steps/os/linux.rs]: Doesn't take into account `system`?

File: src/steps/os/linux.rs [hunk 6]
```diff
@@ -1172,6 +1172,46 @@ pub fn run_cinnamon_spices_updater(ctx: &ExecutionContext) -> Result<()> {
...
+    if ctx.run_type().dry() {
+        println!("Dry running plasmoid-updater");
+        return Ok(());
+    }
+
+    match update(&config) {
+        Err(e) => {
+            println!(
+                "{}",
+                t!("plasmoid-updater encountered an unexpected error during updating:")
```
Comment (GideonBear) [inline src/steps/os/linux.rs:1243]: These are not in the locale file yet. Why doesn't CI complain??

File: Cargo.toml (hunk 2 above)
Comment (uwuclxdy) [reply Cargo.toml]: printing to console from library crate is gated with cli feature (if someone else wished to implement a frontend for it, like a gui or something)

Comment (GideonBear) [reply Cargo.toml]: Oh in that way, it's not "this is a cli" but "this can be used for a cli". Got it!

File: src/steps/os/linux.rs (hunk 3 above)
Comment (uwuclxdy) [reply src/steps/os/linux.rs]: doesn't build without this on other targets (check CI run from previous commit(s))

File: src/steps/os/linux.rs (hunk 5 above)
Comment (uwuclxdy) [reply src/steps/os/linux.rs]: wdym? `ctx.config().yes(Step::Plasmoids) || ctx.config().yes(Step::PlasmoidsSystem)`

Comment (uwuclxdy) [reply src/steps/os/linux.rs]: ah, i see the issue, will fix

File: src/steps/os/linux.rs (hunk 3 above)
Comment (uwuclxdy) [reply src/steps/os/linux.rs]: nvm you're right

File: src/steps/os/linux.rs [hunk 7]
```diff
@@ -2,12 +2,13 @@ use std::path::{Path, PathBuf};
 
 use color_eyre::eyre::Result;
 use ini::Ini;
+use libplasmoid_updater::{update, Config as PlasmoidConfig};
```
Comment (GideonBear) [inline src/steps/os/linux.rs]: eh actually can you just use `libplasmoid_updater::update` directly? Global `update` is not good

**Review by GideonBear — CHANGES_REQUESTED** [review]

File: src/steps/os/linux.rs (hunk 7 above)
Comment (uwuclxdy) [reply src/steps/os/linux.rs]: makes sense
